busy_beaver/adapters/youtube.py

```diff
@@ -0,0 +1,57 @@
+"""Youtube Adapter.
+
+Docs (TODO: Move into documentation)
+
+Where to get your channel id?
+
+Login to youtube and go to https://www.youtube.com/account_advanced, your
+channel id will be displayed.
+
+How to create an api key?
+
+Go to https://console.developers.google.com and create a project. After you can
+visit https://developers.google.com/youtube/registering_an_application#Create_API_Keys
+for instructions to generate the api key. After you have both, you can move to
+the examples.
+
+Example:
+
+api_key = "..."
+channel_id = "..."
+youtube = YoutubeAdapter(api_key=api_key)
+data = youtube.get_latest_videos_from_channel(channel_id)
+videos = data["items"]
+"""
+from typing import Dict
+
+import requests
```

Comment:
I created a layer of abstraction around the request library for JSON APIs. Utilizing this adapter implements sessions under the hood and returns a `Response` namedtuple which makes life easy. The GitHubAdapter uses this pattern.

My next PR includes a beefed up `internal Requests client` with a `post` method and ability to pass in default headers when creating the client. Should be out within a few hours.

https://github.com/alysivji/busy-beaver/blob/master/busy_beaver/adapters/requests_client.py